### ISSUE TITLE
Feedback/Suggestion for new Remote Client

### ISSUE THREAD

**Issue by mbruel** [clementine-player/Clementine#6818]
Hi,

I'm developing a new Remote in C++/Qt/QML so it should be available on any OS (desktop or mobile).
The main goal was to add the missing Files tab from the Android Remote that I need to play my music which is quite tidy in my Music folder and a total mess in the Library.
I'll probably also add the Internet Tab, so we can from the Remote trigger internet radios.
Some change in Clementine player code are needed. I'm doing them on[ this fork](https://github.com/mbruel/Clementine). 
For now I've just patched Clementine to add a **Files root folder** for the Remote Client, and start digging all the protobuf communication. ([cf this thread](https://github.com/clementine-player/Android-Remote/issues/204#issuecomment-699942768))

[Here is a small recording of the how the GUI would look like.](https://files.fm/f/h46f8ueh) 
it's running on Debian and actually connecting to Clementine but I've not started the main views even if I'm getting the playlists and list of songs already.

Basically on top, you have the Menu button followed by the 5 Views: PlayList, Files, Library, Internet, Search 
On the bottom:
  - first row: Download track button, then current track position (slider), then repeat and shuffle buttons
  - second row: buttons to navigate in the active playlist followed by the sound slider.

Any feedback? Something missing? 
All suggestions are welcome. Better to have them as soon as possible when the project is at its start ;)

(Of course I'll try to make the sliders nicer and different for the track position and the sound one. But this will be for later, I'm new to QML...)

**Comment by mbruel** [comment]
I've tweaked a little bit the GUI to display the duration of the track, the time position in the track and the volume level.
![](https://fv2-1.failiem.lv/thumb_show.php?i=ab98p2yd&view&download_checksum=2fef958f4ca05c6f6be23f2347c28fc54161b749&download_timestamp=1603199025)

**Comment by GRbit** [comment]
I would really appreciate the ability to download a playlist / album / artist function

**Comment by mbruel** [comment]
hum... I'll see what I can do and how...
I'm not 100% sure to really develop the Library part myself... it's not my priority as I don't need it. I guess it is from there that you would download an album or an artist.
For the playlist, I keep it in mind, it shouldn't be a problem.
I'll definitely add the possibility to create Playlists, someone asked me for it. I'm not sure why it is not present yet in the Android remote.
I'll keep you updated it here on my progress. And as soon as I'll have a clean and workable version with at least the playlist, I'll publish my work on my github, so we'll switch there ;) it should be by next week-end :p

**Comment by meeotch** [comment]
The features I like the most about the current Android remote are:

- volume button on phone controls clementine volume
- when waking the phone up from sleep, the remote reconnects quickly (like 1 sec)
- doesn't seem to increase battery drain at all to leave it running

Other neat existing features that aren't necessarily dealbreakers:  Optionally lower clementine volume when in a phone call.  Shows currently playing album on phone lock screen.

**Comment by mbruel** [comment]
Alright guys, 
I've just published the first version so you can see the design. 
Only the Playlist View is implemented. It should work with the current version of Clementine.
The project is on my github: [ClementineRemote](https://github.com/mbruel/ClementineRemote)
I've just built a [linux AppImage](https://github.com/mbruel/ClementineRemote/releases/download/v0.1/ClementineRemote_v0.1-x86_64.AppImage) for now. 
I might build macOS or Windows on request but for Mobile, I'll wait to have a more complete version.

**Comment by mbruel** [comment]
@meeotch I only plan to make a generic version that will be deployed the same on any OS. so it won't lower the volume when you get a call. I suppose that when it will be in background the connection will close but the reconnection will be fast.
It won't display on the lock phone.

**Comment by mbruel** [comment]
little update, it's working great on iOS. tested on an old iPhone SE.
I'm finishing the Files view and I'll consider making the first release in the Apple Store
![](https://raw.githubusercontent.com/mbruel/ClementineRemote/main/pics/ClementineRemote_iOS.png)

**Comment by dtantono** [comment]
@mbruel that is wonderful, I hope we can find this app soon.
